tools/grab_spotify_saved_tracks.py

```
@@ -0,0 +1,175 @@
+            with open(cache_path, 'r') as f:
+                token_info = f.read()
+                # In a GitHub Action, you might want to set this as an output
+                # to save for the next run
+                print(f"::set-output name=spotify_token::{token_info}")
+        
+        return sp
+    except Exception as e:
+        logging.error(f"Failed to initialize Spotify client: {e}")
+        raise
+
+def get_saved_tracks(sp, limit=50):
```

Ruff (0.8.2)

Missing return type annotation for public function `get_saved_tracks`

(ANN201)